### PR TITLE
Pass pool_kwargs rather than updating connection_pool_kw

### LINKED ISSUE #3633: HTTPS requests through proxies in proposed/3.0.0 aren't configured correctly
In current master:

```
>>> import requests
>>> requests.__version__
'2.11.1'
>>> session = requests.Session()
>>> r = session.get('https://www.jcline.org/', verify=True, proxies={'http': 'http://vagrant:vagrant@localhost:3128', 'https': 'http://vagrant:vagrant@localhost:3128'})
>>> 
```

In current proposed/3.0.0:

```
>>> import requests
>>> requests.__version__
'3.0.0'
>>> session = requests.Session()
>>> r = session.get('https://www.jcline.org/', verify=True, proxies={'http': 'http://vagrant:vagrant@localhost:3128', 'https': 'http://vagrant:vagrant@localhost:3128'})
requests/packages/urllib3/connectionpool.py:838: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/security.html
  InsecureRequestWarning)
>>> 
```

This is a problem I introduced in https://github.com/kennethreitz/requests/pull/3109 :disappointed:. What happens right now is if a request is _not_ through a proxy and it's HTTPS, the urllib3 pool manager's `connection_pool_kw` are updated before requesting a new connection using [requests.adapters.HTTPAdapter._update_poolmanager_ssl_kw](https://github.com/kennethreitz/requests/blob/proposed/3.0.0/requests/adapters.py#L204). If it _is_ through a proxy, the keywords aren't updated and the request is made with the default settings for urllib3.

To me, the most appealing way to fix this is to add a keyword argument, `connection_kwargs` or something, to all the `urllib3.poolmanager.PoolManager.connection_from_*` methods that is either merged into `connection_pool_kw` or overrides them. That way `urllib3` can handle getting the connection pool with the new kwargs in a thread-safe manner. Currently, `requests` has to manage updating the keys and getting the new connection pool with a lock. It seems like that would be better in `urllib3`.

The other option is to patch up what's currently in `HTTPAdapter` so it handles updating the proxy manager or plain pool manager based on whether proxies are in use.

What do people think?

### PR DESCRIPTION
**Pull request by jeremycline**
This addresses an issue where making HTTPS through proxies used the
default urllib3 connection pool settings.

fixes #3633

### CODE REVIEW

File: requests/adapters.py [hunk 1]
```diff
@@ -204,11 +198,13 @@ def proxy_manager_for(self, proxy, **proxy_kwargs):
 
         return manager
 
-    def _update_poolmanager_ssl_kw(self, verify, cert):
-        """Update the :class:`PoolManager <urllib3.poolmanager.PoolManager>`
-        connection_pool_kw with the necessary SSL configuration. This method
-        should not be called from user code, and is only exposed for use when
-        subclassing the :class:`HTTPAdapter <requests.adapters.HTTPAdapter>`.
+    @staticmethod
+    def _pool_kwargs(verify, cert):
```
Comment (Lukasa) [inline requests/adapters.py]: Don't have static methods: they're just not very useful. Either we should decide users can't override this, in which case this should be a function, or we should decide they can and it becomes an instance method.

Comment (jeremycline) [reply requests/adapters.py]: I can't think of a particularly good reason for someone to want to override this, so a function sounds fine me. If someone really objects to the behaviour for some unforeseen reason, they can override ``get_connection`` and build the pool kwargs their own way.

**Comment by jeremycline** [comment]
This is only a problem in the proposed/3.0.0 branch so even though this breaks the API that's okay, right?

File: requests/adapters.py (hunk 1 above)
Comment (Lukasa) [reply requests/adapters.py]: Yeah, that works for me. Function is the way to go then!

**Comment by Lukasa** [comment]
Yup, no problem at all.

**Comment by jeremycline** [comment]
I have made it so.

**Comment by Lukasa** [comment]
Failing tests. =)

**Comment by jeremycline** [comment]
Oh yeah, that's Python 2.6 because I used a dictionary comprehension in the tests. I can switch it to not use a comprehension, I just thought 2.6 was getting dropped in 3.0.0 and couldn't resist the opportunity to use them :smile:

**Comment by Lukasa** [comment]
heh, right now we can't remove them from the test matrix, so you might have to just tolerate not having it. :grin:

**Comment by jeremycline** [comment]
It's left me with a heavy heart, but the dictionary comprehension has been excised and tests pass on Python 2.6 again.

**Pull request merged**